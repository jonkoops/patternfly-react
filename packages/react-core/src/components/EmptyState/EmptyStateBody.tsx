import { HTMLProps, ReactNode, FunctionComponent } from 'react';
import { css } from '@patternfly/react-styles';
import styles from '@patternfly/react-styles/css/components/EmptyState/empty-state';

export interface EmptyStateBodyProps extends HTMLProps<HTMLDivElement> {
  /** Content rendered inside the empty state body */
  children?: ReactNode;
  /** Additional classes added to the empty state body */
  className?: string;
}

export const EmptyStateBody: FunctionComponent<EmptyStateBodyProps> = ({
  children,
  className,
  ...props
}: EmptyStateBodyProps) => (
  <div className={css(styles.emptyStateBody, className)} {...props}>
    {children}
  </div>
);
EmptyStateBody.displayName = 'EmptyStateBody';
