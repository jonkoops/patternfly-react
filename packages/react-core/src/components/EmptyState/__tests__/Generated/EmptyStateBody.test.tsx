/**
 * @vitest-environment jsdom
 */
import { render } from '@testing-library/react';
import React from 'react';
import { expect, it } from 'vitest';

import { EmptyStateBody } from '../../EmptyStateBody';

it('EmptyStateBody should match snapshot (auto-generated)', () => {
  const { asFragment } = render(<EmptyStateBody children={<div>ReactNode</div>} className={"''"} />);
  expect(asFragment()).toMatchSnapshot();
});
