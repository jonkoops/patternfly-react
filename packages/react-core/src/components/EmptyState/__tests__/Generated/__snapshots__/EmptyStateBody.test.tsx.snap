// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EmptyStateBody should match snapshot (auto-generated) 1`] = `
<DocumentFragment>
  <div
    class="pf-v5-c-empty-state__body ''"
  >
    <div>
      ReactNode
    </div>
  </div>
</DocumentFragment>
`;
